client/app/views/folder.coffee

```diff
@@ -219,121 +218,118 @@ module.exports = class FolderView extends BaseView
             @$('#files-drop-zone').hide()
 
+
     onDrop: (event) ->
         event.preventDefault()
         event.stopPropagation()
         return false if @isPublic and not @canUpload
 
-        # folder drag and drop is only supported in Chrome
-        if event.dataTransfer.items?
-            @onFilesSelectedInChrome event
-        else
-            @onFilesSelected event
+        @onFilesSelected event
 
         @uploadButton.removeClass 'btn-cozy-contrast'
         @$('#files-drop-zone').hide()
 
+
     onDirectorySelected: (event) ->
         input = @$ '#folder-uploader'
         files = input[0].files
         return unless files.length
+
         @uploadQueue.addFolderBlobs files, @model
+
         # reset the input
         input.replaceWith input.clone true
 
-    onFilesSelected: (event) =>
-        files = event.dataTransfer?.files or event.target.files
-
-        if files.length
-            @uploadQueue.addBlobs files, @model
-
-
-    onFilesSelectedInChrome: (e) ->
-        items = e.dataTransfer.items
-        return unless items.length
 
+    onFilesSelected: (event={}) =>
         files = []
         errors = []
 
-        # Once all entries have been parsed, they are added to the upload queue.
-        addAllToQueue = =>
-
-            processUpload = =>
-                @uploadQueue.addFolderBlobs files, @model
-
-            if errors.length > 0
-                formattedErrors = errors
-                    .map (name) -> "\"#{name}\""
-                    .join ', '
-                localeOptions =
-                    files: formattedErrors
-                    smart_count: errors.length
+        _saveEntry = (entry, path='', next) ->
+            entry.file (file) ->
+                relativePath = "#{path}/#{file.name}"
+                file.relativePath = relativePath
+                file.webkitRelativePath = relativePath
+                files.push file
+                next()
+
+            , (error) ->
+                errors.push entry.name
+                next()
+
+        _addDirectories = (items, path='/', callback) ->
+            async.eachSeries items, (entry, next) ->
+                # 1rst case : event data type
+                # otherwhise its a recursive call
+                if entry instanceof DataTransferItem
+                    entry = entry.webkitGetAsEntry()
+
+                # Save file
+                if entry.isFile
+                    _saveEntry entry, path, next
+
+                # Save directory
+                # and its subfiles and/or subdirectory
+                else if entry.isDirectory
+                    entry.createReader().readEntries (entries) ->
+                        path = entry.fullPath.substring 1
+                        _addDirectories entries, path, next
+
+            , (args...) ->
+                # Save all files or goto nextDirectory
+                # it depends on recursive or not call
+                callback() if callback?
+
+
```

Comment:
I think it could be better to be more accurate here with a comment like 
```
# we check 'items' property (if exists) because 
# this is the only way, for now, to get the difference 
# between files and folder drag and drop
# using isFile() or isDirectory()
```